Makefile

```diff
@@ -64,93 +93,113 @@ __check_defined = \
     $(if $(value $1),, \
       $(error Undefined $1$(if $2, ($2))))
 
-help:
-	./gradlew help $(params)
-
 clean:
 	rm -rf `find -type d -name build`
 
-publish_to_maven_local:
-	./gradlew publishToMavenLocal -PprojectVersion=local $(log_level)
+# Warning. Hack!
+# Мы можем удалять эти локи, т.к. гарантированно никакие другие процессы не используют этот шаренный кеш на начало новой сборки
+# см. clearDockerContainers
+# То что лок файлы остаются от предыдущих сборок, означает что мы где-то неправильно останавливаем процесс
+# '|| true' необходим для свеже-поднятых агентов, где еще не создана папка с кешами
+clear_gradle_lock_files:
+	find "$(GRADLE_HOME_DIR)" \( -name "*.lock" -o -name "*.lck" \) -delete || true
+
+# По-разным причинам работа контейнера при прошлой сборке может не завершиться
+# Здесь мы перестраховываемся и останавливаем все работающие контейнеры
+# Перед сборкой не должно быть других контейнеров в любом случае
+clear_docker_containers:
+	containers=$(docker container ls -aq)
+	if [[ ! -z "$(containers)" ]]; then \
+	docker container rm --force $(containers); \
+	fi
+
+init_git:
```

Comment:
I'd like to hide such functions from autocompletion.
As for me, they are internal.

At least, we can add prefix to show this.

Reply:
agree

Reply:
moved it to entrypoint